### ISSUE TITLE
Implement long-term Stackage strategy

### ISSUE THREAD

**Issue by dogonthehorizon** [Nike-Inc/hal#39]
As soon as some out-of-bounds packages got added to Stackage the build failed for `hal` and it was removed from nightly :(

In `0.1.2` we added `pvp-bounds: both-revision` so that `stack sdist && stack upload` don't initially set bounds and we don't have to spend as much time fixing build errors.

That said, we still need to add more LTS/nightly versions of the lib to upcoming Travis builds so we make sure we're aware of any changes as early as possible.

**Comment by JackKelly-Bellroy** [comment]
Is this relevant? CI appears to have included stackage nightly for a while.

**Comment by IamfromSpace** [comment]
> CI appears to have included stackage nightly for a while.

Yep, wrangling `envy` was the final piece of getting this in order.  From here it's ready to go to get back into Stackage.

**Comment by IamfromSpace** [comment]
Absolutely, Been a bit pressed for time lately, but it's on my list.  I'm excited to get it dropped from the `extra-deps` of a few projects :)
